src/main/scala/code.scala

```diff
@@ -4,18 +4,52 @@ import ohnosequences.loquat._
 import ohnosequences.statika._
 import ohnosequences.cosas._, types._, klists._
 import ohnosequences.datasets._
+import play.api.libs.ws._
+import play.api.libs.ws.ahc._
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration.Duration
+import scala.concurrent.ExecutionContext.Implicits.global
+import java.nio.file.Files.newOutputStream
 
 // awful name
 case object code {
 
-  // TODO implement, add API client whatever
-  // should return the same file in CheckedFile if OK
+  // TODO: Compute checksum (md5, whatever)
+  val checkFile:
+    File => CheckedFile =
+      ???
+
+  val openStream:
+    File => FileStream =
+      file => (file, newOutputStream(file.toPath))
+
   val downloadTo:
-    BasespaceAPI  =>
-    File          =>
-    BasespaceURL  =>
-    BasespaceError + CheckedFile =
-    ???
+    WSClient     =>
+    BasespaceURL =>
+    FileStream   =>
+    (WSClient, FileStream, Future[BasespaceError + File]) =
```

Comment:
for what we want to do here returning `Future` won't help in any way. Better to get rid of it ASAP